hack/generated/apis/microsoft.resources/v1alpha1api20200601/resourcegroup_types.go

```diff
@@ -181,6 +240,42 @@ func (spec *ResourceGroupSpec) PopulateFromARM(owner genruntime.KnownResourceRef
 // SetAzureName sets the Azure name of the resource
 func (spec *ResourceGroupSpec) SetAzureName(azureName string) { spec.AzureName = azureName }
 
+func (spec *ResourceGroupSpec) ConvertSpecTo(destination genruntime.ConvertibleSpec) error {
```

Comment:
These are more like `AssignSpecTo` and `AssignSpecFrom` really, aren't they? In the non-handwritten versions of these methods, are `source` and `destination` guaranteed to be the same type as the receiver?

Reply:
In the generated versions, we have both `AssignProperties<To|From>()` and `ConvertSpec<To|From>()`. Given that we have exactly one version of ResourceGroup and that's not likely to change, I combined the functionality together here. 

Do you think I should maintain the same pattern and split them out?

To answer your specific question, the `AssignProperties<To|From>()` methods have strongly typed parameters of the exact type, but of the next version of the resource. So the pattern isn't exactly the same.

Reply:
Ok, that makes sense - the part I was missing is that there's only one version of ResourceGroup and so this is the only reasonable way to implement `ConvertSpecTo` et al. Maybe add a comment to that effect?